boxsdk/util/api_call_decorator.py

```diff
@@ -0,0 +1,53 @@
+# coding: utf-8
+
+from __future__ import unicode_literals, absolute_import
+
+from functools import update_wrapper
+
+
+def api_call(func):
+    """
+    Designates the decorated method as one that makes a Box API call.
+    The decorated method can then accept a new keyword argument `request_data`,
+    a dictionary of key-value pairs to be passed to the network layer for API
+    calls made by the method.
+
+    :param func:
+        The method to decorate.
```

Comment:
This needs to be a method of a `Cloneable` subclass. Document that explicitly, and maybe also make that clearer in the param name. Perhaps instead of `func` (which might make one believe that I can decorate anything with it), we can use `method`.